docs/eventing/experimental-features/new-trigger-filters.md

```diff
@@ -0,0 +1,161 @@
+
+```yaml=
+filters:
+  - suffix:
+      type: .created
+```
+
+### `all`
+
+All nested filter expessions must evaluate to true.
+
+```yaml=
+filters:
+  - all:
+      - exact:
+          type: com.github.push
+      - exact:
+          subject: https://github.com/cloudevents/spec
+```
+
+### `any`
+
+At least one nested filter expession must evaluate to true.
+
+```yaml=
+filters:
+  - any:
+      - exact:
+          type: com.github.push
+      - exact:
+          subject: https://github.com/cloudevents/spec
+```
+
+### `not`
+
+The nested expression evaluated must evaluate to false.
+
+```yaml=
+filters:
+  - not:
+      - exact:
+          type: com.github.push 
+```
+### `sql`
+
+The provided [CloudEvents SQL Expression](https://github.com/cloudevents/spec/blob/main/cesql/spec.md) must evaluate to true.
+
+```yaml=
+filters:
+  - sql: "source LIKE '%commerce%' AND type IN ('order.created', 'order.updated', 'order.canceled')"
+```
+
+## Conflict with the current `filter` field
+
+The current `filter` field will continue to be supported. However, when the feature is enabled and an object includes both `filter` and `filters`, the new `filters` field will override the `filter` field. This will allow you to try out the effect of the new `filters` field without compromising existing filters and you can introduce it to existing `Trigger` objects gradually.
+
+```yaml=
+apiVersion: eventing.knative.dev/v1
+kind: Trigger
+metadata:
+  name: my-service-trigger
+spec:
+  broker: default
+  # Current filter field. Will be ignored.
+  filter:
+    attributes:
+      type: dev.knative.foo.bar
+      myextension: my-extension-value
+  # Enhanced filters field. This will override the old filter field.
+  filters:
+    - sql: "type == 'dev.knative.foo.bar' AND myextension == 'my-extension-value'"
+  subscriber:
+    ref:
+      apiVersion: serving.knative.dev/v1
+      kind: Service
+      name: my-service
+```
+## FAQ
+
+### Why are we adding yet another field? Why make the current `filter` field more robust?
+
+The reason there is twofold. First, at the time of developing `Trigger` APIs, there was no Subscriptions API in CloudEvents Project, so it makes sense to experiment with an API that makes us closer to Subscriptions API. Second, we still want to support users workload with the old `filter` field, and give them the possibility to transition to the new `filters` field.
+
+### Why `filters` and not another name that wouldn't be conflicting with the `filter` field?
+
+We've considered naming it `cefilters`, `subscriptionsAPIFilters` or `enhancedFilters` or something less conflicting. But we decided that introducing yet another different name than what's in the Subscriptions API is a step further from the alignment path we want to take towards the Subscriptions API. Instead, we decided it is a good opportunity to conform with the Subscriptions API at least starting from the field name level and leveraging the safety of this being an experimental feature.
```

Comment:
```suggestion
We considered other names, such as `cefilters`, `subscriptionsAPIFilters`, or `enhancedFilters`, but we decided that this would be a step further from aligning with the Subscriptions API. Instead, we decided it is a good opportunity to conform with the Subscriptions API, at least at the field name level, and to leverage the safety of this being an experimental feature.
```